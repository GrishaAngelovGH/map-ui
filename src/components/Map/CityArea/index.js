import CityArea from './CityArea'

export default CityArea