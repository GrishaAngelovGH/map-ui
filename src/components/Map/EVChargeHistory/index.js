import EVChargeHistory from './EVChargeHistory'

export default EVChargeHistory