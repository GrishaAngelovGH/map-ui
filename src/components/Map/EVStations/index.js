import EVStations from './EVStations'

export default EVStations