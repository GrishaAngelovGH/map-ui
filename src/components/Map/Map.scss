.react-leaflet-map {
  width: 100%;
  height: 100vh;
}

@media screen and (min-width: 992px) {
  .react-leaflet-map {
    width: 100%;
    height: calc(100vh - 4rem);
  }
}
