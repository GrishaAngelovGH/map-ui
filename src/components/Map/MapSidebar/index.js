import MapSidebar from './MapSidebar'

export default MapSidebar