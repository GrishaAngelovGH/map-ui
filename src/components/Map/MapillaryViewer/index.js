import MapillaryViewer from './MapillaryViewer'

export default MapillaryViewer