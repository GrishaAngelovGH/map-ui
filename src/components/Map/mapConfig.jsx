import MarkerIcon from './MarkerIcon'

export const map = {
  lat: 51.505,
  lng: -0.09,
  zoom: {
    default: 13,
    min: 10,
    max: 16
  },
  url: 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png'
}

export const motionPolylineConf = locations => {
  return L.motion.polyline(locations, {
    color: 'lime',
    weight: 10
  }, {
    auto: true,
    duration: 15000,
    speed: 5,
    easing: L.Motion.Ease.linear
  }, {
    removeOnEnd: false,
    showMarker: true,
    icon: MarkerIcon('suv')
  })
}