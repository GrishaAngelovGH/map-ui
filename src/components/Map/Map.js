import { useState, useRef } from 'react'
import { MapContainer, TileLayer, FeatureGroup } from 'react-leaflet'

import Locations from './Locations'
import MapButton from './MapButton'
import TrafficReport from './TrafficReport'
import VehicleRoute from './VehicleRoute'

import 'leaflet/dist/leaflet.css'
import './Map.scss'

const map = {
  lat: 51.505,
  lng: -0.09,
  zoom: {
    default: 13,
    min: 12,
    max: 16
  },
  url: 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png'
}

const Map = ({ showLocations, showRoute, resetControls }) => {
  const [showTrafficReport, setTrafficReport] = useState(false)
  const featureGroupRef = useRef()

  const handleShowTrafficReport = () => {
    setTrafficReport(!showTrafficReport)
  }

  const handleClearMap = () => {
    resetControls()
    featureGroupRef.current.clearLayers()
    featureGroupRef.current._map.setZoom(map.zoom.default)
  }

  return (
    <MapContainer
      center={[map.lat, map.lng]}
      zoom={map.zoom.default}
      minZoom={map.zoom.min}
      maxZoom={map.zoom.max}
      className='react-leaflet-map'
    >
      <TileLayer url={map.url} />

      <MapButton
        icon='layout-text-sidebar-reverse'
        title='Traffic Report'
        onClick={handleShowTrafficReport}
      />

      <MapButton
        icon='backspace'
        title='Clear Map'
        onClick={handleClearMap}
      />

      <FeatureGroup ref={featureGroupRef}>
        {
          showLocations && <Locations />
        }

        {
          showTrafficReport && (
            <TrafficReport
              show={showTrafficReport}
              onHide={handleShowTrafficReport}
            />
          )
        }

        {showRoute && <VehicleRoute />}
      </FeatureGroup>
    </MapContainer>
  )
}

export default Map