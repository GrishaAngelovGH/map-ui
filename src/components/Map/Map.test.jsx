import { render } from '@testing-library/react'

import Map from './Map'

test('should render Map component', () => {
  const view = render(<Map />)

  expect(view).toMatchSnapshot()
})