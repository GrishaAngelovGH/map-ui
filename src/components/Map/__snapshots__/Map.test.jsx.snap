// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Map component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="react-leaflet-map leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="left: 0px; top: 0px;"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; left: 0px; top: 0px;"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  src="https://b.tile.openstreetmap.org/13/4093/2724.png"
                  style="width: 256px; height: 256px; left: -244px; top: -35px;"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          />
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          >
            <div
              class="leaflet-control-zoom leaflet-bar leaflet-control"
            >
              <a
                aria-disabled="false"
                aria-label="Zoom in"
                class="leaflet-control-zoom-in"
                href="#"
                role="button"
                title="Zoom in"
              >
                <span
                  aria-hidden="true"
                >
                  +
                </span>
              </a>
              <a
                aria-disabled="false"
                aria-label="Zoom out"
                class="leaflet-control-zoom-out"
                href="#"
                role="button"
                title="Zoom out"
              >
                <span
                  aria-hidden="true"
                >
                  −
                </span>
              </a>
            </div>
          </div>
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          >
            <div
              class="leaflet-control-attribution leaflet-control"
            >
              <a
                href="https://leafletjs.com"
                title="A JavaScript library for interactive maps"
              >
                Leaflet
              </a>
            </div>
          </div>
        </div>
        <div
          class="row g-0 justify-content-center align-items-center"
        >
          <div
            class="col-10 col-md-4 bg-map-menu rounded border border-2 shadow p-2 d-flex justify-content-center"
          >
            <div
              class="btn-group"
              role="group"
            >
              <button
                class="btn btn-outline-success"
                title="Underground Locations"
                type="button"
              >
                <img
                  class=""
                  src="/src/images/map/map-menu/button-group/ldn-underground.png"
                  width="40"
                />
              </button>
              <button
                class="btn btn-outline-success"
                title="London Boroughs"
                type="button"
              >
                <img
                  class=""
                  src="/src/images/map/map-menu/button-group/map.png"
                  width="40"
                />
              </button>
              <button
                class="btn btn-outline-success"
                title="EV Stations"
                type="button"
              >
                <img
                  class=""
                  src="/src/images/map/map-menu/button-group/charging-station.png"
                  width="40"
                />
              </button>
              <button
                class="btn btn-outline-success"
                title="Street View"
                type="button"
              >
                <img
                  class=""
                  src="/src/images/map/map-menu/button-group/street-view.png"
                  width="40"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="leaflet-right"
        >
          <div
            class="leaflet-control mt-2"
          >
            <button
              class="btn btn-light"
              title="Traffic Report"
            >
              <i
                class="bi bi-layout-text-sidebar-reverse"
              />
            </button>
          </div>
        </div>
        <div
          class="leaflet-right"
        >
          <div
            class="leaflet-control mt-2"
          >
            <button
              class="btn btn-light"
              title="Play Route"
            >
              <i
                class="bi bi-play-circle"
              />
            </button>
          </div>
        </div>
        <div
          class="leaflet-right"
        >
          <div
            class="leaflet-control mt-2"
          >
            <button
              class="btn btn-light"
              title="City Area"
            >
              <i
                class="bi bi-hexagon"
              />
            </button>
          </div>
        </div>
        <div
          class="leaflet-right"
        >
          <div
            class="leaflet-control mt-2"
          >
            <button
              class="btn btn-light"
              title="EV Charge History"
            >
              <i
                class="bi bi-ev-station"
              />
            </button>
          </div>
        </div>
        <div
          class="leaflet-right"
        >
          <div
            class="leaflet-control mt-2"
          >
            <button
              class="btn btn-light"
              title="Clear Map"
            >
              <i
                class="bi bi-backspace"
              />
            </button>
          </div>
        </div>
        <div
          class="leaflet-right"
        >
          <div
            class="leaflet-control mt-2"
          >
            <button
              class="position-fixed top-50 start-0 sidebar-map-button"
            >
              <i
                class="bi bi-chevron-right"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="react-leaflet-map leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
      style="position: relative;"
      tabindex="0"
    >
      <div
        class="leaflet-pane leaflet-map-pane"
        style="left: 0px; top: 0px;"
      >
        <div
          class="leaflet-pane leaflet-tile-pane"
        >
          <div
            class="leaflet-layer "
            style="z-index: 1;"
          >
            <div
              class="leaflet-tile-container leaflet-zoom-animated"
              style="z-index: 18; left: 0px; top: 0px;"
            >
              <img
                alt=""
                class="leaflet-tile"
                src="https://b.tile.openstreetmap.org/13/4093/2724.png"
                style="width: 256px; height: 256px; left: -244px; top: -35px;"
              />
            </div>
          </div>
        </div>
        <div
          class="leaflet-pane leaflet-overlay-pane"
        />
        <div
          class="leaflet-pane leaflet-shadow-pane"
        />
        <div
          class="leaflet-pane leaflet-marker-pane"
        />
        <div
          class="leaflet-pane leaflet-tooltip-pane"
        />
        <div
          class="leaflet-pane leaflet-popup-pane"
        />
      </div>
      <div
        class="leaflet-control-container"
      >
        <div
          class="leaflet-top leaflet-left"
        >
          <div
            class="leaflet-control-zoom leaflet-bar leaflet-control"
          >
            <a
              aria-disabled="false"
              aria-label="Zoom in"
              class="leaflet-control-zoom-in"
              href="#"
              role="button"
              title="Zoom in"
            >
              <span
                aria-hidden="true"
              >
                +
              </span>
            </a>
            <a
              aria-disabled="false"
              aria-label="Zoom out"
              class="leaflet-control-zoom-out"
              href="#"
              role="button"
              title="Zoom out"
            >
              <span
                aria-hidden="true"
              >
                −
              </span>
            </a>
          </div>
        </div>
        <div
          class="leaflet-top leaflet-right"
        />
        <div
          class="leaflet-bottom leaflet-left"
        />
        <div
          class="leaflet-bottom leaflet-right"
        >
          <div
            class="leaflet-control-attribution leaflet-control"
          >
            <a
              href="https://leafletjs.com"
              title="A JavaScript library for interactive maps"
            >
              Leaflet
            </a>
          </div>
        </div>
      </div>
      <div
        class="row g-0 justify-content-center align-items-center"
      >
        <div
          class="col-10 col-md-4 bg-map-menu rounded border border-2 shadow p-2 d-flex justify-content-center"
        >
          <div
            class="btn-group"
            role="group"
          >
            <button
              class="btn btn-outline-success"
              title="Underground Locations"
              type="button"
            >
              <img
                class=""
                src="/src/images/map/map-menu/button-group/ldn-underground.png"
                width="40"
              />
            </button>
            <button
              class="btn btn-outline-success"
              title="London Boroughs"
              type="button"
            >
              <img
                class=""
                src="/src/images/map/map-menu/button-group/map.png"
                width="40"
              />
            </button>
            <button
              class="btn btn-outline-success"
              title="EV Stations"
              type="button"
            >
              <img
                class=""
                src="/src/images/map/map-menu/button-group/charging-station.png"
                width="40"
              />
            </button>
            <button
              class="btn btn-outline-success"
              title="Street View"
              type="button"
            >
              <img
                class=""
                src="/src/images/map/map-menu/button-group/street-view.png"
                width="40"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="leaflet-right"
      >
        <div
          class="leaflet-control mt-2"
        >
          <button
            class="btn btn-light"
            title="Traffic Report"
          >
            <i
              class="bi bi-layout-text-sidebar-reverse"
            />
          </button>
        </div>
      </div>
      <div
        class="leaflet-right"
      >
        <div
          class="leaflet-control mt-2"
        >
          <button
            class="btn btn-light"
            title="Play Route"
          >
            <i
              class="bi bi-play-circle"
            />
          </button>
        </div>
      </div>
      <div
        class="leaflet-right"
      >
        <div
          class="leaflet-control mt-2"
        >
          <button
            class="btn btn-light"
            title="City Area"
          >
            <i
              class="bi bi-hexagon"
            />
          </button>
        </div>
      </div>
      <div
        class="leaflet-right"
      >
        <div
          class="leaflet-control mt-2"
        >
          <button
            class="btn btn-light"
            title="EV Charge History"
          >
            <i
              class="bi bi-ev-station"
            />
          </button>
        </div>
      </div>
      <div
        class="leaflet-right"
      >
        <div
          class="leaflet-control mt-2"
        >
          <button
            class="btn btn-light"
            title="Clear Map"
          >
            <i
              class="bi bi-backspace"
            />
          </button>
        </div>
      </div>
      <div
        class="leaflet-right"
      >
        <div
          class="leaflet-control mt-2"
        >
          <button
            class="position-fixed top-50 start-0 sidebar-map-button"
          >
            <i
              class="bi bi-chevron-right"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
